napari/components/viewer_model.py

```diff
@@ -843,6 +846,9 @@ def open(
         stack: bool = False,
         plugin: Optional[str] = None,
         layer_type: Optional[str] = None,
+        select_reader_helper: Callable[
```

Comment:
for such an important part of our public API, it would be great if we can can come up with a solution that doesn't expose a relatively confusing parameter.  Would any napari user need to use this?  or is this just here as an internal hook that we pretty much want everyone to ignore?